### ISSUE TITLE
Legit website blocked: www.emarketer.com

### ISSUE THREAD

**Issue by Quorum75** [StevenBlack/hosts#1715]
Reading my news when I found out that https://www.emarketer.com was blocked. This is a legit website about e-commerce, retail... news

 ![image](https://user-images.githubusercontent.com/49731083/126759914-e79e0fcb-dd4f-404b-82d5-368c907f26e2.png)

**Comment by dnmTX** [comment]
**ping Peter** @pgl for assist 👍 
Pete,just FYI. You probably had good reason to list it but by not including the `www.` part i'm still able to access the website.

**Comment by Quorum75** [comment]
> **ping Peter** @pgl for assist 👍 
> 
> Pete,just FYI. You probably had good reason to list it but by not including the `www.` part i'm still able to access the website.

That's very strange because I still can't...

![image](https://user-images.githubusercontent.com/49731083/126818007-1ea1a43b-8ff3-4ca8-972b-d2412a52eef8.png)

![image](https://user-images.githubusercontent.com/49731083/126818020-44e92ee1-558a-462a-8345-f774a507139d.png)

**Comment by Quorum75** [comment]
You know what, I just did a quick search in 
https://raw.githubusercontent.com/StevenBlack/hosts/master/alternates/fakenews-gambling-porn-social/hosts
And I couldn't find it. So, I'm kind of confused now...

**Comment by dnmTX** [comment]
> However I think I heard somewhere there was a bug in iOS 14, not sure if it's true though

iPhone's browser(Safari) allways adds `www.` for some reason.It's been like that for a very long time.
Same with Chrome/Chromium. If i block `0.0.0.0 example.com` but the actual site is `www.example.com` i'll still be able to access it. Thats why the curators have to follow the exact name of the website due to hosts formats don't provide wildcard blocking.

**Comment by Quorum75** [comment]
FYI the full link I was trying to get access to this morning was:

https://www.emarketer.com/content/walmart-is-an-ecommerce-winner-during-pandemic?utm_content=bufferb1a77&utm_medium=social&utm_source=linkedin.com&utm_campaign=buffer

**Comment by dnmTX** [comment]
Opens just fine on my end and i have `0.0.0.0 emarketer.com` listed in my hosts file:

![Capture](https://user-images.githubusercontent.com/34774426/126826271-4df19db3-7fbd-4335-ab73-e8282cb07cf2.PNG)

**Comment by krystian3w** [comment]
I suppose failed to win "bataille royale".

**Comment by Quorum75** [comment]
> > However I think I heard somewhere there was a bug in iOS 14, not sure if it's true though
> 
> 
> 
> iPhone's browser(Safari) allways adds `www.` for some reason.It's been like that for a very long time.
> 
> Same with Chrome/Chromium. If i block `0.0.0.0 example.com` but the actual site is `www.example.com` i'll still be able to access it. Thats why the curators have to follow the exact name of the website due to hosts formats don't provide wildcard blocking.

Yes that is why I use Adguard Pro as a second layer of security. 
AG syntax allow you to block with only one rule "example.com and www.example.com" including its subdomains.
Then, it also helps to reduce the size of the DNS rules.

**Comment by pgl** [comment]
> **ping Peter** @pgl for assist 👍
> Pete,just FYI. You probably had good reason to list it but by not including the `www.` part i'm still able to access the website.

Hey. So, in my list I only include the highest-level part of a domain that seems useful. I think it's already been explained about the use of exact matches for `hosts` files so I won't go into detail again - but just to say that this is the case for basically every parent domain in my list.

**Comment by dnmTX** [comment]
Pete @pgl i'm not asking you to include it if it's not present but when it is present. For example:
**Not present(initially):**
![not](https://user-images.githubusercontent.com/34774426/126862784-a25d1ecb-96c2-4898-9147-b8f12b9ebce3.PNG)
**Present:**
![present](https://user-images.githubusercontent.com/34774426/126862792-72cb176e-6be5-4d35-9f33-05250aa8942b.PNG)

So...if for some reason you decide to block google as `0.0.0.0 google.com` i'll still be able to access it. 
But if you do `0.0.0.0 www.google.com` i and basically no one else will definitely not.
I hope you got the point.

**Comment by pgl** [comment]
@dnmTX OK, but my list isn't really a `hosts` file. It's just a list of domains, and I offer it in `hosts` file format. I could add a `www.` prefix to every entry, but it would double the size for download. Maybe I could add another [format option](https://pgl.yoyo.org/as/formats.php) I guess, one with the `www.` and the usual one without. I just haven't done this since I started.

I think the best thing would be to request anyone incorporating it to add the `www.` prefix if they're including it in their blocklist.

**Comment by dnmTX** [comment]
@pgl don't do any of it,it's wrong. Most of the domains just don't have it's own additional version which starts with `www.` 
and simply will not respond,aka dead weight.
Just....remember for the feature from now on.Whenever and however you see the domain you're planning to include in your list,
if,then that same domain has `www.` DON'T skip it and list it whithout it(it will not be blocked,believe me).
If `www.` is not present,just list it as it is and that's all,it shouldn't be hard,right? 😉

**Comment by pgl** [comment]
@dnmTX I think I completely understand what you're saying, but I'm not going to include two versions of a domain in my list (with, and without the prefix), sorry. I add the highest part of a domain that I think is relevant.

**Comment by dnmTX** [comment]
> @dnmTX I think I completely understand what you're saying...

You didn't apparently,and just for the record i never asked you to include two versions,just the opposite but i guess we won't get on the same page and i'm really not here to fight so will leave it at that. No point to continue further.
Good night.

**Comment by StevenBlack** [comment]
So this might be a problem.

I don't think `emarketer.com` and `www.emarketer.com` should be blocked unless it's shown their root domains are shown to be vectors.  This is a business and our role isn't to deplatform businesses, just their shady vectors.

> but I'm not going to include two versions of a domain in my list (with, and without the prefix), sorry. I add the highest part of a domain that I think is relevant.

Peter @pgl I count 1,354 of 3,600+ domains in the `yoyo.org` list contain subdomains.  So I understand what you're saying, but I'm having difficulty squaring that with the evidence.  

Regardless, `emarketer.com` shouldn't be blocked in the first place, unless Peter @pgl you've got reasons we can't see.

**Comment by dnmTX** [comment]
> Regardless, emarketer.com shouldn't be blocked in the first place

It's not(unless you using wildcard blocking,aka NextDNS or whatever....) due to root domain is actually `www.emarketer.com`.

**Comment by StevenBlack** [comment]
Thanks Dan, I understand what you're saying, and I disagree.

The root domain is, by definition, `emarketer.com`.  `www.emarketer.com` is, by definition, a subdomain.

Maybe you're really saying, `emarketer.com` isn't responding to port 80 (http) or port 443 (https)?  This http/https functionality is separate and distinct from the notion of a root domain.

A hosts file will block a domain lookup regardless of which port is used...

**Comment by dnmTX** [comment]
Steve @StevenBlack it's not about port(s) or whatever. It's about how that domain has been registered in the first place.
Because even if you type in your browser `emarketer.com`(but that same domain is registerd as `www.emarketer.com`) the browser
will recieve response: "Well....we don't have `emarketer.com` but we do have `www.emarketer.com`,would you like that instead?
You see where i'm going with this?
Bottom line....sub-domain or not it has to be listed as it's appear when first seen,and if `www.` was present then,it has to be included.Not as a additional entry but just as that same individual one.

**Comment by krystian3w** [comment]
I suppose iOS no have implemented `badfilter`: 

https://kb.adguard.com/en/general/how-to-create-your-own-ad-filters#badfilter-modifier

So harder is unlock load page without allow load ads/tracking.

**Comment by dnmTX** [comment]
Steve @StevenBlack can't you see that you can freely browse `www.emarketer.com` on your end even though you're using the hosts file and `0.0.0.0 emarketer.com` is listed there.
Doesn't that ring the bell that in it's current form(`0.0.0.0 emarketer.com`) is useless.

**Comment by dnmTX** [comment]
@krystian3w  It's time for you to GO. **TROLL** time is OVER ❗

**Comment by StevenBlack** [comment]
Dan @dnmTX look, I get it, OK?  I'm not under any illusions about this.  And I know how all this works.

They aren't serving a website, possibly not exposing any server functionality at all, on `emarketier.com`.  That doesn't mean it's "useless".  It means they have ops people that probably know a thing or two, because configuring like this is a choice.

`emarketer.com` is exposed as an A-record on DNS.

My sole point here is, we have a problem here because i haven't sees any evidence that `www.emarketer.com` should be listed.

Please be nice to our visitors.  I think it's probably better to ignore snark than snark-back.

**Comment by krystian3w** [comment]
In uBO possible use:

`||emarketer.com^$badfilter` to unlock `www.` but iOS content blocker API is very limited.

**Comment by dnmTX** [comment]
> My sole point here is, we have a problem here because i haven't seen any evidence that www.emarketer.com should be listed. And it's not.

I'd say it's time to close that issue then because it was about domain that...as you said...it's not listed here(i fully agree on that part)
👍

**Comment by StevenBlack** [comment]
Dan @dnmTX we still have the notion that Peter's @pgl list has a root-domain bent.  That doesn't quite align with what I aspire this list to be.

Blocking root domains of businesses, without cause, is a recipe for lawyers incoming.  I'll fight like hell for cause, but I don't want to block business websites without cause.

Peter's not blocking `google.com`.  Nor should he, in my view.  But that somehow falls apart with smaller businesses and that's conceptually problematic for me.

**Comment by krystian3w** [comment]
Still visible as "mirroring":

https://github.com/StevenBlack/hosts/blob/377994090740ff22a15551aa4b1a40dc16e5375c/data/yoyo.org/hosts#L1720

IMO fight with remove FP form PGL is pointless.

**Comment by dnmTX** [comment]
> Dan @dnmTX we still have the notion that Peter's @pgl list has a root-domain bent.

I'll leave you two to figure that part out 😉 👍

**Comment by pgl** [comment]
@StevenBlack Heya. OK so, if I see a domain that's being used for tracking purposes, then I'll add it to the list. `emarketer.com` is a very old entry - it was added Sat Aug 28th 2004 08:20. It's possible things have changed since I added it.

(The `www.` prefix is a special case, so as a rule I would strip it and add the parent domain. I think that's a side issue though.)

I'm not out to ban whole businesses, but if it's a tracking company then adding their domain is usually easiest to (try to) block their tracking activities. For people who want to go directly to their site and have a legit use, then most of the time they're not using a tracking blocker in the first place, or if they are then it's a bit ironic that they want to be delisted. But even so I'm happy to replace parent domain entries with sets of subdomains if they provide me with a list that are used for tracking / ad purposes.

It's kind of late right now, but I'll take another look at this site later and see if I can figure out what the domains are used for tracking. I don't want to block a legit site so it looks like I'll probably remove the entry.

**Comment by StevenBlack** [comment]
Thanks Peter @pgl.  I'm going to close this now, I'm happy.  Keep up the good fight brother.

**Comment by pgl** [comment]
I've replaced `emarketer.com` with `analytics.emarketer.com`, and also added `api.amplitude.com` as I noticed they're using that on their site.

@StevenBlack You too mate. :)

**Comment by ler762** [comment]
@pgl do you also block cname cloaking sites?

```
;; ANSWER SECTION:
on.emarketer.com.       86400   IN      CNAME   emarketer.mktoweb.com.
emarketer.mktoweb.com.  300     IN      CNAME   sj05.mktossl.com.
sj05.mktossl.com.       300     IN      A       104.17.72.206


;; ANSWER SECTION:
hello.emarketer.com.    86400   IN      CNAME   mkto-sj050105.com.
mkto-sj050105.com.      300     IN      A       104.17.70.206
```

**Comment by Quorum75** [comment]
> @StevenBlack Heya. OK so, if I see a domain that's being used for tracking purposes, then I'll add it to the list. `emarketer.com` is a very old entry - it was added Sat Aug 28th 2004 08:20. It's possible things have changed since I added it.
> 
> 
> 
> (The `www.` prefix is a special case, so as a rule I would strip it and add the parent domain. I think that's a side issue though.)
> 
> 
> 
> I'm not out to ban whole businesses, but if it's a tracking company then adding their domain is usually easiest to (try to) block their tracking activities. For people who want to go directly to their site and have a legit use, then most of the time they're not using a tracking blocker in the first place, or if they are then it's a bit ironic that they want to be delisted. But even so I'm happy to replace parent domain entries with sets of subdomains if they provide me with a list that are used for tracking / ad purposes.
> 
> 
> 
> It's kind of late right now, but I'll take another look at this site later and see if I can figure out what the domains are used for tracking. I don't want to block a legit site so it looks like I'll probably remove the entry.

Emarketer.com isn't a tracker company but a research one and belongs to Business Insider. It is the Gartner for CMO. 

"For nearly 25 years, eMarketer has been trusted by CMOs as the most comprehensive source of information on how to operate in a digital world, offering transparently sourced and vetted data from thousands of sources distilled into charts, forecasts, and robust reports that provide unparalleled insight into changing consumer behaviors."

**Comment by StevenBlack** [comment]
@Quorum75 I've said this before, and I want this to be clear.

I don't care if a domain belongs to the Holy Sisters relict of Mother Theresa of Calcutta. If the domain employs scum practices, or vehicles crapware or worse, they get listed here.

So what you quote above doesn't enter.  Not at all.

Just so we're clear.

**Comment by Quorum75** [comment]
No issue, then you may want to inform yourself before talking. If we cannot work we will just stop to use your host and will avoid your usual aggressiveness and rudeness. 
Get some anger management therapy mate.
